plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java

```diff
@@ -170,6 +170,39 @@ public void testCreateAndDropTableWithSpecialCharacterName(String tableName)
         assertFalse(getQueryRunner().tableExists(getSession(), tableName));
     }
 
+    // TODO remove this test after https://github.com/trinodb/trino/issues/14517
+    @Test(dataProvider = "testTableNameDataProvider")
+    public void testRenameColumnNameAdditionalTests(String columnName)
+    {
+        String nameInSql = "\"" + columnName.replace("\"", "\"\"") + "\"";
+        String tableName = "tcn_" + nameInSql.replaceAll("[^a-z0-9]", "") + randomTableSuffix();
+        // Use complex identifier to test a source column name when renaming columns
+        String sourceColumnName = "a;b$c";
+
+        assertUpdate("CREATE TABLE " + tableName + "(\"" + sourceColumnName + "\" varchar(50))");
+        assertTableColumnNames(tableName, sourceColumnName);
+
+        assertUpdate("ALTER TABLE " + tableName + " RENAME COLUMN \"" + sourceColumnName + "\" TO " + nameInSql);
+        assertTableColumnNames(tableName, columnName.toLowerCase(ENGLISH));
+
+        assertUpdate("DROP TABLE " + tableName);
```

Comment:
Could we use TestTable here?

Reply:
this test is a copy from BCT which will be removed after https://github.com/trinodb/trino/issues/14517 will be done,
so I prefer to keep it the same as in BCT, if we want some additional checks we can add it in 
https://github.com/trinodb/trino/issues/14517,
or I didn't understand correctly your comment

Reply:
I mean that maybe
```
assertUpdate("CREATE TABLE " + tableName + "(\"" + sourceColumnName + "\" varchar(50))");
...
assertUpdate("DROP TABLE " + tableName);
```
can be replaced with something like
```
try (TestTable table = ...) {
...
}
```

Reply:
we shouldn't use testtabel, since we alter the table name in the meantime

Reply:
We don't rename table for `testRenameColumnName...`
and for `testRenameFromToTableWithSpecialCharacterName` there is 
> // test rename back is working properly

Reply:
> and for `testRenameFromToTableWithSpecialCharacterName` there is
> 
> > // test rename back is working properly

i know we rename it _back_, but still using `TestTable` for something the test manages explicitly is not OK

Reply:
An example of when it can cause issues is if the test gets modified over time and then TestTable (if a custom other than `io.trino.testing` is used) starts leaking tables.

Reply:
Agree about testing table rename